Assets/MRTK/SDK/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -44,33 +45,112 @@ public MigrationTool()
         /// Adds selectedObject to the list of objects to be migrated. Return false if the object is not of type GameObject, or SceneAsset.
         /// </summary>
         /// <param name="selectedObject">Unity Object of type GameObject or SceneAsset</param>
-        public bool TryAddObjectForMigration(Object selectedObject)
+        public bool TryAddObjectForMigration(Type type, Object selectedObject)
         {
+            if(type == null)
+            {
+                Debug.LogError("Migration type needs to be selected before migration.");
+                return false;
+            }
+
+            if(type != migrationHandlerInstanceType)
+            {
+                ClearMigrationList();
+                Debug.LogWarning("New migration type selected for migration. Clearing previous selection.");
```

Comment:
```suggestion
                Debug.Log("New migration type selected for migration. Clearing previous selection.");
```

Reply:
Nit: I don't think this is really a warning type thing right. It's more just informational